frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { RefObject } from "react";
+import {
+  Box,
+  Table,
+  TableContainer,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@chakra-ui/react";
+import { Resident } from "../../../types/ResidentTypes";
+import getFormattedDateAndTime from "../../../utils/DateUtils";
+
+type Props = {
+  residents: Resident[];
+  tableRef: RefObject<HTMLDivElement>;
+};
+
+const getFormattedDatesAndStatus = (resident: Resident) => {
+    const startDateObj = new Date(resident.dateJoined);
+    const startDate = getFormattedDateAndTime(startDateObj, true);
+
+    let endDate;
+    if (resident.dateLeft != null) {
+        const endDateObj = new Date(resident.dateLeft)
+        endDate = getFormattedDateAndTime(endDateObj, true)
+    }
+    const status = resident.dateJoined !== null && resident.dateLeft !== null ? "Past" : "Current";
+    return {
+        startDate,
+        endDate,
+        status
+    }
+}
+
+const ResidentDirectoryTable = ({
+  residents,
+  tableRef,
+}: Props): React.ReactElement => {
+  return (
+    <Box>
+      <TableContainer
+        marginTop="12px"
+        height="70vh"
+        overflowY="scroll"
+        ref={tableRef}
+      >
+        <Table
+          variant="showTable"
+          minHeight="400px"
+          verticalAlign="middle"
+        >
+          <Thead>
+            <Tr>
+              <Th>Resident</Th>
+              <Th>Status</Th>
+              <Th>Building</Th>
+              <Th>Residency Start Date</Th>
+              <Th>Residency End Date</Th>
+            </Tr>
+          </Thead>
+
+          <Tbody>
+            {residents.map((resident) => {
+              const residentId = resident.initial + resident.roomNum;
```

Comment:
Hm I thought this was apart of the resident table? cc @Safewaan

Reply:
yeah it is, I merged my fix so rebase main and update this

Reply:
@kevin-pierce tl;dr get residents pulls the resident id so you don't have to do this step

Reply:
Resolved ; just updating the type as well in this PR

Reply:
So there seems to be a bit of discrepancy between the `Resident` type we use for creation + fetching ; the residents we fetch from our DB have the `residentId` field, but the type involved in `addResident` is different in that it _does not need to supply this field__  ; as a temporary solution to expedite this work, I've just used an non-null assertion on the `residentId` field, and I'll resolve the type issues separately

Reply:
Created - https://github.com/uwblueprint/supportive-housing/issues/148 to track this